linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -575,88 +575,84 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
-          }
-
-        case Assign(select @ ArraySelect(array, index), rhs) =>
-          unnest(array, index, rhs) { (newArray, newIndex, newRhs, env0) =>
-              implicit val env = env0
-              val genArray = transformExprNoChar(newArray)
-              val genIndex = transformExprNoChar(newIndex)
-              val genRhs = transformExpr(newRhs, select.tpe)
-              semantics.arrayIndexOutOfBounds match {
-                case CheckedBehavior.Compliant | CheckedBehavior.Fatal =>
-                  js.Apply(js.DotSelect(genArray, js.Ident("set")),
-                      List(genIndex, genRhs))
-                case CheckedBehavior.Unchecked =>
-                  js.Assign(
-                      js.BracketSelect(
-                          js.DotSelect(genArray, js.Ident("u"))(select.pos),
-                          genIndex)(select.pos),
-                      genRhs)
+        case Assign(lhs, rhs) =>
+          lhs match {
+            case Select(qualifier, className, field) =>
+              unnest(qualifier, rhs) { (newQualifier, newRhs, env0) =>
+                implicit val env = env0
+                js.Assign(
+                    genSelect(transformExprNoChar(newQualifier), className, field)(lhs.pos),
+                    transformExpr(newRhs, lhs.tpe))
               }
-          }
-
-        case Assign(lhs: RecordSelect, rhs) =>
-          val newLhs = Transient(JSVarRef(makeRecordFieldIdentForVarRef(lhs),
-              mutable = true)(lhs.tpe))
-          pushLhsInto(Lhs.Assign(newLhs), rhs, tailPosLabels)
 
-        case Assign(select @ JSPrivateSelect(qualifier, className, field), rhs) =>
-          unnest(qualifier, rhs) { (newQualifier, newRhs, env0) =>
-            implicit val env = env0
-            js.Assign(
-                genJSPrivateSelect(transformExprNoChar(newQualifier),
-                    className, field)(moduleContext, globalKnowledge, select.pos),
-                transformExprNoChar(newRhs))
-          }
+            case ArraySelect(array, index) =>
+              unnest(array, index, rhs) { (newArray, newIndex, newRhs, env0) =>
+                  implicit val env = env0
+                  val genArray = transformExprNoChar(newArray)
+                  val genIndex = transformExprNoChar(newIndex)
+                  val genRhs = transformExpr(newRhs, lhs.tpe)
+                  semantics.arrayIndexOutOfBounds match {
+                    case CheckedBehavior.Compliant | CheckedBehavior.Fatal =>
+                      js.Apply(js.DotSelect(genArray, js.Ident("set")),
+                          List(genIndex, genRhs))
+                    case CheckedBehavior.Unchecked =>
+                      js.Assign(
+                          js.BracketSelect(
+                              js.DotSelect(genArray, js.Ident("u"))(lhs.pos),
+                              genIndex)(lhs.pos),
+                          genRhs)
+                  }
+              }
```

Comment:
There's an indentation issue here:
```suggestion
              unnest(array, index, rhs) { (newArray, newIndex, newRhs, env0) =>
                implicit val env = env0
                val genArray = transformExprNoChar(newArray)
                val genIndex = transformExprNoChar(newIndex)
                val genRhs = transformExpr(newRhs, lhs.tpe)
                semantics.arrayIndexOutOfBounds match {
                  case CheckedBehavior.Compliant | CheckedBehavior.Fatal =>
                    js.Apply(js.DotSelect(genArray, js.Ident("set")),
                        List(genIndex, genRhs))
                  case CheckedBehavior.Unchecked =>
                    js.Assign(
                        js.BracketSelect(
                            js.DotSelect(genArray, js.Ident("u"))(lhs.pos),
                            genIndex)(lhs.pos),
                        genRhs)
                }
              }
```